teuthology/openstack/__init__.py

```diff
@@ -1062,6 +1062,10 @@ def get_user_data(self):
         if self.args.simultaneous_jobs:
             all_options += [ '--nworkers ' + str(self.args.simultaneous_jobs) ]
         if self.args.nameserver:
+            try:
+                socket.inet_aton(self.args.nameserver)
+            except socket.error:
+                raise Exception("nameserver must be a valid IP address")
```

Comment:
1) This change is not related to the PR.
2) In general name server should not be a valid IP only.

Reply:
Wouldn't it be crazy to specify the nameserver as a hostname, though?

Would you specify a hostname for the nameserver in /etc/resolv.conf ?

Reply:
Get hostname ip before you need to set IP?